foundation/org.eclipse.persistence.core/src/org/eclipse/persistence/mappings/AggregateMapping.java

```diff
@@ -354,6 +354,39 @@ public ChangeRecord compareForChange(Object clone, Object backup, ObjectChangeSe
         return changeRecord;
     }
 
+    public static Set<java.lang.reflect.Field> getAllFields(final Set<java.lang.reflect.Field> fields, final Class<?> type) {
+        final java.lang.reflect.Field[] declaredFields = type.getDeclaredFields();
+        for (final java.lang.reflect.Field declaredField : declaredFields) {
+            if (!java.lang.reflect.Modifier.isStatic(declaredField.getModifiers())) {
+                fields.add(declaredField);
+            }
+        }
+
+        if (type.getSuperclass() != null) {
+            getAllFields(fields, type.getSuperclass());
+        }
+
+        return fields;
+    }
+
+    public static boolean checkNull(final Object o) {
+        if (o == null) {
+            return true;
+        }
+        final Set<java.lang.reflect.Field> fields = getAllFields(new HashSet<java.lang.reflect.Field>(), o.getClass());
+        for (final java.lang.reflect.Field f : fields) {
```

Comment:
If `Object o` is not null and contains no fields, then return `true`?

Reply:
Yes, that is intended as there should not be any difference. Do you see any problem in that behavior?